api/usersendpoint.go

```diff
@@ -34,28 +40,44 @@ type UsersResponse struct {
 }
 
 func UsersHandler(ctx echo.Context) error{
+	const (
+		dateFormat = "20060102"
```

Comment:
Since this date format is the same across everything, it should only be defined once. If you make a separate file with validation functions as I mention elsewhere, that might be the best single place to put this constant.

One other question I'd have is: why not allow specifying a more exact timestamp? The UI can always simplify down to day-level resolution (in which case, _it_ would handle the "add a day" logic I've talked about elsewhere), and it maybe makes this service more broadly useful if needed in the future. It also might make the code simpler, particularly if you can just use the `time.RFC3339Nano` constant or something like that. (This is what I'm doing in the async-tasks service I'm working on at present, though that's much more internal of course).

Reply:
I made it down to the day because that is just as specific as the UI date pickers are. It did not occur to me that someone could query by the hour / minute. Not sure I understand how to set that up though

Reply:
So, the rough outline of how that would work:

* here in the service, you'd use `time.RFC3339Nano` everywhere you're currently using your custom date format. That format looks like: `"2006-01-02T15:04:05.999999999Z07:00"`. Basically, it allows a complete timestamp down to absolutely unnecessary levels of precision, including timezone (those 9s mean that you don't need to include all 9 digits of the sub-second precision, though). You'd parse it as you are, but not adding any days or anything -- since it specifies a very precise point in time, you should pass it directly down to the DB.
* Since it includes a timezone, you'd also make sure the DB knows about the timezone (specifically, that it translates it into the appropriate `timestamp without time zone` to compare to the database column values). I do this in the [async-tasks service I've been working on lately](https://github.com/ianmcorvidae/async-tasks/blob/1ad1dd943b104fb2d859f52c6b6a0330430381b1/database.go#L439) by way of using postgresql's configured timezone, and you should be able to do something similar.
* if the UI wants day-level precision, it handles turning the days into the precise timestamps. So while right now it would get a date picker with, say, `2019-12-01` to `2019-12-02` and pass those to the service as `20191201` and `20191202`, it would instead pass them as `2019-12-01T00:00:00-07:00` and `2019-12-03T00:00:00-07:00` (noting that `-07:00` is the timezone for AZ, which we tend to use everywhere, and noting that the UI, rather than the service, changes `2019-12-02` into `2019-02-03` at midnight).

For now, I think leaving the UI with day-level precision is good, but making the service more flexible means that it's more convenient in the future if we want to change things.